### PR TITLE
OCD-4207 - Developer URL Uptime report based on existing CHPL URL monitors

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-resources/src/main/resources/system-triggers.xml [hunk 1]
```diff
@@ -197,5 +197,15 @@
...
             </cron>
         </trigger>
+        <trigger>
+            <cron>
+                <name>generateUrlUptime</name>
+                <group>generateUrlUptimeTrigger</group>
+                <job-name>urlUptimeCreatorJob</job-name>
+                <job-group>systemJobs</job-group>
+                <misfire-instruction>MISFIRE_INSTRUCTION_DO_NOTHING</misfire-instruction>
+                <cron-expression>0 0 5 * * ?</cron-expression> <!-- At 1045 UTC on the second Saturday of the month -->
```
Comment (kekey1) [inline chpl/chpl-resources/src/main/resources/system-triggers.xml]: Just noting the comment here doesn't seem to match the cron expression

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-resources/src/main/resources/email.properties [hunk 2]
```diff
@@ -353,6 +353,12 @@ developer.missingAttestationChangeRequest.report.subject=Semi-annual Attestation
 developer.missingAttestationChangeRequest.report.salutation=Hello %s,
 developer.missingAttestationChangeRequest.report.paragraph=Emails were sent to the following developers:
 
+#URL Uptime Report
+urlUptime.report.filename=URL Uptime
+urlUptime.report.subject=CHPL URL Uptime Report
+urlUptime.report.heading=CHPL URL Uptime Report
+urlUptime.report.paragraph1.body=See attached file
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/email.properties]: Should we have more content here? Maybe not needed for this ticket, but could be worth floating the idea to get it on the PO's radar

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/UrlUptimeCalculator.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                .pastWeekTestCount(Long.valueOf(allTestsForPastWeek.size()))
+                .pastWeekSuccessfulTestCount(allTestsForPastWeek.stream()
+                        .filter(test -> test.getPassed())
+                        .count())
+                .build();
+    }
+
+    private List<ChplUptimeMonitorTest> getEligibleTestsForPastWeek(List<ChplUptimeMonitorTest> chplUptimeMonitorTests) {
+        LocalDateTime end = LocalDateTime.now().minusDays(1).with(ChronoField.NANO_OF_DAY, LocalTime.MAX.toNanoOfDay());
+        LocalDateTime start = LocalDateTime.now().minusDays(6).with(ChronoField.NANO_OF_DAY, LocalTime.MIN.toNanoOfDay());
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/UrlUptimeCalculator.java]: Should this be?

```suggestion
        LocalDateTime start = LocalDateTime.now().minusDays(7).with(ChronoField.NANO_OF_DAY, LocalTime.MIN.toNanoOfDay());
```

I can see that you're going to the end of the day yesterday on line 68, and the beginning of the day on line 69, but I think it needs to go back one day further to have a full seven days of data

**Review by andlar — COMMENTED** [review]
We'd talked, a while ago, about trying to write unit tests that would cover some of the work that happens in our quartz job code stuff. Is there any value of trying to do that here?

**Review by kekey1 — DISMISSED** [review]
Approve but I see a number of checkstyle warnings about trailing spaces in the jobs.xml file, log4j2 files, and email.properties for things we are newly adding. I think these should be corrected.

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**